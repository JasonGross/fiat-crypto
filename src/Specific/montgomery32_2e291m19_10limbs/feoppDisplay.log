opp = λ var : Z.Syntax.base_type → Type,
λ '(x16, x17, x15, x13, x11, x9, x7, x5, x3, x1)%core,
uint32_t x19, uint8_t x20 = subborrow_u32(0x0, 0x0, x1);
uint32_t x22, uint8_t x23 = subborrow_u32(x20, 0x0, x3);
uint32_t x25, uint8_t x26 = subborrow_u32(x23, 0x0, x5);
uint32_t x28, uint8_t x29 = subborrow_u32(x26, 0x0, x7);
uint32_t x31, uint8_t x32 = subborrow_u32(x29, 0x0, x9);
uint32_t x34, uint8_t x35 = subborrow_u32(x32, 0x0, x11);
uint32_t x37, uint8_t x38 = subborrow_u32(x35, 0x0, x13);
uint32_t x40, uint8_t x41 = subborrow_u32(x38, 0x0, x15);
uint32_t x43, uint8_t x44 = subborrow_u32(x41, 0x0, x17);
uint32_t x46, uint8_t x47 = subborrow_u32(x44, 0x0, x16);
uint32_t x48 = cmovznz32(x47, 0x0, 0xffffffff);
uint32_t x50, uint8_t x51 = addcarryx_u32(0x0, x19, (x48 & 0xffffffed));
uint32_t x53, uint8_t x54 = addcarryx_u32(x51, x22, (x48 & 0xffffffff));
uint32_t x56, uint8_t x57 = addcarryx_u32(x54, x25, (x48 & 0xffffffff));
uint32_t x59, uint8_t x60 = addcarryx_u32(x57, x28, (x48 & 0xffffffff));
uint32_t x62, uint8_t x63 = addcarryx_u32(x60, x31, (x48 & 0xffffffff));
uint32_t x65, uint8_t x66 = addcarryx_u32(x63, x34, (x48 & 0xffffffff));
uint32_t x68, uint8_t x69 = addcarryx_u32(x66, x37, (x48 & 0xffffffff));
uint32_t x71, uint8_t x72 = addcarryx_u32(x69, x40, (x48 & 0xffffffff));
uint32_t x74, uint8_t x75 = addcarryx_u32(x72, x43, (x48 & 0xffffffff));
uint32_t x77, uint8_t _ = addcarryx_u32(x75, x46, ((uint8_t)x48 & 0x7));
(Return x77, Return x74, Return x71, Return x68, Return x65, Return x62, Return x59, Return x56, Return x53, Return x50)
     : ∀ var : Z.Syntax.base_type → Type, expr Z.Syntax.base_type Z.Syntax.op (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t -> uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)

Argument scope is [function_scope]
