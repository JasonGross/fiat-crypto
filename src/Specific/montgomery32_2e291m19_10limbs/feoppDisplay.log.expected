λ x : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32,
Interp-η
(λ var : Syntax.base_type → Type,
 λ '(x17, x18, x16, x14, x12, x10, x8, x6, x4, x2)%core,
 uint32_t x20, uint8_t/*bool*/ x21 = subborrow_u32(0x0, 0x0, x2);
 uint32_t x23, uint8_t/*bool*/ x24 = subborrow_u32(x21, 0x0, x4);
 uint32_t x26, uint8_t/*bool*/ x27 = subborrow_u32(x24, 0x0, x6);
 uint32_t x29, uint8_t/*bool*/ x30 = subborrow_u32(x27, 0x0, x8);
 uint32_t x32, uint8_t/*bool*/ x33 = subborrow_u32(x30, 0x0, x10);
 uint32_t x35, uint8_t/*bool*/ x36 = subborrow_u32(x33, 0x0, x12);
 uint32_t x38, uint8_t/*bool*/ x39 = subborrow_u32(x36, 0x0, x14);
 uint32_t x41, uint8_t/*bool*/ x42 = subborrow_u32(x39, 0x0, x16);
 uint32_t x44, uint8_t/*bool*/ x45 = subborrow_u32(x42, 0x0, x18);
 uint32_t x47, uint8_t/*bool*/ x48 = subborrow_u32(x45, 0x0, x17);
 uint32_t x49 = cmovznz32(x48, 0x0, 0xffffffff);
 uint32_t x50 = (x49 & 0xffffffed);
 uint32_t x52, uint8_t/*bool*/ x53 = addcarryx_u32(0x0, x20, x50);
 uint32_t x54 = (x49 & 0xffffffff);
 uint32_t x56, uint8_t/*bool*/ x57 = addcarryx_u32(x53, x23, x54);
 uint32_t x58 = (x49 & 0xffffffff);
 uint32_t x60, uint8_t/*bool*/ x61 = addcarryx_u32(x57, x26, x58);
 uint32_t x62 = (x49 & 0xffffffff);
 uint32_t x64, uint8_t/*bool*/ x65 = addcarryx_u32(x61, x29, x62);
 uint32_t x66 = (x49 & 0xffffffff);
 uint32_t x68, uint8_t/*bool*/ x69 = addcarryx_u32(x65, x32, x66);
 uint32_t x70 = (x49 & 0xffffffff);
 uint32_t x72, uint8_t/*bool*/ x73 = addcarryx_u32(x69, x35, x70);
 uint32_t x74 = (x49 & 0xffffffff);
 uint32_t x76, uint8_t/*bool*/ x77 = addcarryx_u32(x73, x38, x74);
 uint32_t x78 = (x49 & 0xffffffff);
 uint32_t x80, uint8_t/*bool*/ x81 = addcarryx_u32(x77, x41, x78);
 uint32_t x82 = (x49 & 0xffffffff);
 uint32_t x84, uint8_t/*bool*/ x85 = addcarryx_u32(x81, x44, x82);
 uint8_t x86 = ((uint8_t)x49 & 0x7);
 uint32_t x88, uint8_t/*bool*/ _ = addcarryx_u32(x85, x47, x86);
 (Return x88, Return x84, Return x80, Return x76, Return x72, Return x68, Return x64, Return x60, Return x56, Return x52))
x
     : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 → ReturnType (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)
